### PR TITLE
DEV-86: New MobileModal Component.

### PR DESCRIPTION
**Pull request by VagnerNico**
* Created the MobileComponent following the Figma prototype.

  * The component also has the option to become something like the DraggableBottomSheet. This Component is used on Uber mobile to displays the current service detail.

Done

Closes DEV-86

### CODE REVIEW

File: src/components/MobileModal/MobileModal.tsx [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+            swipeState.isAnimating ||
+            swipeState.isDragging ||
+            swipeState.isOpen ||
+            open
+              ? `block`
+              : `none`,
+          height: `100vh`,
+          left: 0,
+          // Computed opacity to match our text.40
+          opacity: 0.6017699115044248,
```
Comment (giubatt) [inline src/components/MobileModal/MobileModal.tsx:275]: Can't you use `"text.alpha.40"` for the `backgroundColor` in this case? Or is the opacity important for other elements?

File: src/components/MobileModal/MobileModal.tsx [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+enum MobileModalTypes {
+  WITH_OVERFLOW = "overflow-herz-mobile-modal",
+  WITHOUT_OVERFLOW = "herz-mobile-modal",
+}
+
+export interface MobileModalProps {
+  backgroundStyles?: SxStyleProp
+  children?: ReactNode
+  dismissible?: (() => void) | boolean
```
Comment (giubatt) [inline src/components/MobileModal/MobileModal.tsx:22]: It feels a bit confusing to me to have `dismissible` potentially receive a callback or a boolean in terms of the component API.

I think separating it in `dismissible` and `onDismiss` would be more intuitive, what do you think?

File: src/components/MobileModal/MobileModal.tsx (hunk 1 above)
Comment (VagnerNico) [reply src/components/MobileModal/MobileModal.tsx:275]: Actually, the opacity property is required to maintain some visibility on the overlaid elements.

File: src/components/MobileModal/MobileModal.tsx (hunk 2 above)
Comment (VagnerNico) [reply src/components/MobileModal/MobileModal.tsx:22]: Sure, I agree. That property can be very confusing for library users.
I will fix and push into this PR.

File: src/components/MobileModal/MobileModal.tsx (hunk 1 above)
Comment (VagnerNico) [reply src/components/MobileModal/MobileModal.tsx]: Fixed

File: src/components/MobileModal/MobileModal.tsx (hunk 2 above)
Comment (VagnerNico) [reply src/components/MobileModal/MobileModal.tsx]: Fixed

**Review by giubatt — APPROVED** [review]

**Review by welitonderesende — DISMISSED** [review]

**Review by giubatt — APPROVED** [review]

**Review by giubatt — APPROVED** [review]

**Review by welitonderesende — APPROVED** [review]

**Review by welitonderesende — DISMISSED** [review]

**Review by giubatt — DISMISSED** [review]

**Review by giubatt — APPROVED** [review]

**Review by welitonderesende — APPROVED** [review]

**Pull request merged**